### PR TITLE
task: workflow for build docker and deploy image

### PR DESCRIPTION
**Pull request by timhuang1018**

### CODE REVIEW

File: .github/workflows/deploy_image.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+      - name: Retrieve the secret and decode it to a file
+        env:
+          LOCAL_CONFIG: ${{ secrets.LOCAL_CONFIG }}
+        run: |
+          echo $LOCAL_CONFIG | base64 --decode > local.ini
+      - name: Set Release tag to env
+        run: echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
```
Comment (birdca) [inline .github/workflows/deploy_image.yml:23]: could you give me an example of this, what the input and output could be? Thx!

File: .github/workflows/deploy_image.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      - name: Replace version in api.yml
+        run: |
+          sed -i "s/richer1018\/application:[0-9]\+\.[0-9]\+\.[0-9]\+/richer1018\/application:$RELEASE_VERSION/g" api.yml
+      - name: Pull docker image
+        env:
+          DOCKER_HUB_TOKEN: ${{ secrets.DOCKER_HUB_TOKEN }}
+        run: |
+          docker login -u "richer1018" -p $DOCKER_HUB_TOKEN
+          docker pull richer1018/application:$RELEASE_VERSION
```
Comment (birdca) [inline .github/workflows/deploy_image.yml:60]: oh, i though you're going to use the <DOCKER_IMG>:latest to pull the latest docker image

File: .github/workflows/deploy_image.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      - name: Deploy docker image
+        run: docker stack deploy --with-registry-auth -c api.yml api
```
Comment (birdca) [inline .github/workflows/deploy_image.yml:62]: Am i getting it correct? We build api, crawler, scheduler images together in application image and deploy application image through this line?

Comment (timhuang1018) [reply .github/workflows/deploy_image.yml:62]: nope, this PR just deploy the api,  I'm not sure when we will start using crawler, scheduler... into our development and product. Let's wait until it's needed

File: .github/workflows/deploy_image.yml (hunk 1 above)
Comment (timhuang1018) [reply .github/workflows/deploy_image.yml:23]: hmm I think it just create a variable append into whole github environment so other steps could retrieve, seems like won't get output execute this line
see the ref: https://github.com/birdca/Richer/actions/runs/6871230096/job/18687677356

File: .github/workflows/deploy_image.yml (hunk 2 above)
Comment (timhuang1018) [reply .github/workflows/deploy_image.yml:60]: let's use version variable coming from tag for now, since it will be used in other place as well

Comment (birdca) [reply .github/workflows/deploy_image.yml:60]: okay~

File: .github/workflows/deploy_image.yml (hunk 3 above)
Comment (birdca) [reply .github/workflows/deploy_image.yml:62]: got it!

File: .github/workflows/deploy_image.yml (hunk 1 above)
Comment (birdca) [reply .github/workflows/deploy_image.yml:23]: cool, I'm fine with it!

**Pull request merged**